astropy/coordinates/earth.py

```diff
@@ -646,12 +651,45 @@ def get_itrs(self, obstime=None):
 
         # do this here to prevent a series of complicated circular imports
         from .builtin_frames import ITRS
-        return ITRS(x=self.x, y=self.y, z=self.z, obstime=obstime)
+        itrs_coo = ITRS(x=self.x, y=self.y, z=self.z, obstime=obstime)
+        if include_velocity:
+            zeros = np.broadcast_to(0. * u.km / u.s, (3,) + itrs_coo.shape, subok=True)
+            itrs_coo.data.differentials['s'] = CartesianDifferential(zeros)
+        return itrs_coo
 
     itrs = property(get_itrs, doc="""An `~astropy.coordinates.ITRS` object  with
                                      for the location of this object at the
                                      default ``obstime``.""")
 
+    def get_posvel(self, obstime, frame):
+        """
+        Calculate the position and velocity of this object in a given
+        ``frame`` at the requested ``obstime``.
+
+        Parameters
+        ----------
+        obstime : `~astropy.time.Time`
+            The ``obstime`` to calculate the GCRS position/velocity at.
+        frame : class
+            The coordinate frame in which the position and velocity should
+            be measured.
+
+        Returns
+        --------
+        location : `~astropy.coordinates.CartesianRepresentation`
+            The position of the object in the specified ``frame``.
+        velocity : `~astropy.coordinates.CartesianRepresentation`
+            The velocity of the object in the specified ``frame``.
+        """
+        if not inspect.isclass(frame):
+            raise TypeError('frame must be a class')
+
+        frame_instance = frame(obstime=obstime)
+        data = self.get_itrs(obstime=obstime, include_velocity=True).transform_to(frame_instance).data
```

Comment:
here, you'd need `frame.obstime`.  Would one also want to do `.cartesian` instead of `.data`? Then, the differentials get already transformed automatically, and you could just do
```
return data.without_differentials(), data.differentials['s']
```

The brevity to me also somewhat argues against the need for the method...  I'll make a PR to your branch so you can see what I have in mind...

Reply:
However this ends up, I do think it is important for it to be fairly intuitive for the user. For instance, I use `get_gcrs_posvel()` myself in order to form an astrometric position and will continue to do so until an astrometric frame is available.